ui/app/adapters/pki/pki-issuer-engine.js

```diff
@@ -0,0 +1,28 @@
+import ApplicationAdapter from '../application';
+import { encodePath } from 'vault/utils/path-encoding-helpers';
+
+export default class PkiIssuerEngineAdapter extends ApplicationAdapter {
```

Comment:
As mentioned in the Roles PR, we will likely make our own named-path.js type serializer that can be extended for each of these routes adapters. However, at this point, it's helpful to individualize the functionality or extend from current functionality. Later, for a clean up tasks and when we remove the Old PKI engine, we'll clean this up.

Reply:
Right now this code, is pulled out of the secrets backend pki-role adapter, and modified for the issuer endpoint.

Reply:
Not a blocker but would it make sense to extend the adapter that you mentioned rather than copying the code?

Reply:
It's a bit of a toss up. The biggest issue being it's not the exact same code (I should have made that clear). That code was for roles, this is for a new endpoint issuers. It's more or less the same flow, but modified. So It couldn't be extended without modifying the original pki-role adapter.